pkg/ioutils/readers.go

```diff
@@ -189,6 +190,9 @@ func (r *bufReader) drain() {
 		reuseCount++
 		r.wait.Signal()
 		r.Unlock()
+		//allow or force Go scheduler to switch context, without explicitly
+		//forcing this will make it hang when using gccgo implementation
+		runtime.Gosched()
```

Comment:
I really hope we're not going to resort to doing something like this in the future for other parts of the code as well.

gccgo probably has a lot more issues and sprinkling calls to runtime functions all over sounds like a bad way to handle these issues.